# Author: OpenAI Codex
# Keep zip deletion patterns quoted so the shell does not expand them against
# Bazel's execroot before Info-ZIP can match them against jar entries.
diff --git a/third_party/BUILD b/third_party/BUILD
index 1f35568..709a449 100644
--- a/third_party/BUILD
+++ b/third_party/BUILD
@@ -486,16 +486,16 @@ alias(
 )
 
 UNNECESSARY_DYNAMIC_LIBRARIES = select({
-    "//src/conditions:windows": "*.so *.jnilib",
-    "//src/conditions:darwin": "*.so *.dll",
-    "//src/conditions:linux_x86_64": "*.jnilib *.dll",
-    "//src/conditions:linux_s390x": "*.jnilib *.dll",
+    "//src/conditions:windows": "'*.so' '*.jnilib'",
+    "//src/conditions:darwin": "'*.so' '*.dll'",
+    "//src/conditions:linux_x86_64": "'*.jnilib' '*.dll'",
+    "//src/conditions:linux_s390x": "'*.jnilib' '*.dll'",
     # The .so file is an x86/s390x one, so we can just remove it if the CPU is not x86/s390x
-    "//src/conditions:arm": "*.so *.jnilib *.dll",
-    "//src/conditions:linux_aarch64": "*.so *.jnilib *.dll",
-    "//src/conditions:linux_ppc": "*.so *.jnilib *.dll",
-    "//src/conditions:freebsd": "*.so *.jnilib *.dll",
-    "//src/conditions:openbsd": "*.so *.jnilib *.dll",
+    "//src/conditions:arm": "'*.so' '*.jnilib' '*.dll'",
+    "//src/conditions:linux_aarch64": "'*.so' '*.jnilib' '*.dll'",
+    "//src/conditions:linux_ppc": "'*.so' '*.jnilib' '*.dll'",
+    "//src/conditions:freebsd": "'*.so' '*.jnilib' '*.dll'",
+    "//src/conditions:openbsd": "'*.so' '*.jnilib' '*.dll'",
     # Default is to play it safe -- better have a big binary than a slow binary
     #   The empty string means nothing is to be removed from the library;
     #   the rule command tests for the empty string explictly to avoid
@@ -528,7 +528,7 @@ genrule(
     cmd = "cp $< $@ && " +
           # Make sure we can write the output file, even if the input isn't writable.
           "chmod +w $@ && " +
-          "zip -qd $@ */license/* " + UNNECESSARY_DYNAMIC_LIBRARIES,
+          "zip -qd $@ '*/license/*' " + UNNECESSARY_DYNAMIC_LIBRARIES,
 )
 
 distrib_java_import(
